### PR TITLE
Add GUI for supercritical sludge to gas process

### PR DESCRIPTION
**Pull request by luohezhiming**
## Fixes/Resolves:
The previous costing flow for catalyst is based on the wrong flow (inlet flow), I modify to the specific catalyst flow for AT-HTL and HTG

## Summary/Motivation:
Add GUI for supercritical sludge to gas process (AMO1614)

## Changes proposed in this PR:
-add GUI
-fixed the chemical dose flow

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+        input_category="HTG",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.HTG.catalyst_dosage,
+        name="Catalyst dosing",
+        ui_units=pyunits.pound / pyunits.metric_ton,
+        display_units="lb/ton of inlet flow rate",
+        rounding=2,
+        description="Specific power relating the power to inlet mass flow rate",
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]:
```suggestion
        description="Catalyst dosage",
```

Comment (MarcusHolly) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]: Small typo here

**Review by MarcusHolly — APPROVED** [review]

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+
+def export_variables(flowsheet=None, exports=None):
+    fs = flowsheet
+    # --- Input data ---
+    # Feed conditions
+    flow_mass = (
+        fs.feed.flow_mass_comp[0, "H2O"]
+        + fs.feed.flow_mass_comp[0, "organic_solid"]
+        + fs.feed.flow_mass_comp[0, "inorganic_solid"]
+        + fs.feed.flow_mass_comp[0, "organic_solid"]
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]: I am guessing this duplicate was by accident (i.e., you add mass flow for `organic_solid` twice). Is there another component that should be added here or should you just delete the duplicate?

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+        + fs.feed.flow_mass_comp[0, "carbon_dioxide"]
+    )
+    exports.add(
+        obj=flow_mass,
+        name="Volumetric flow rate",
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]:
```suggestion
        name="Mass flow rate",
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+        rounding=1,
+        description="Inlet mass flow rate",
+        is_input=True,
+        input_category="Feed",
+        is_output=True,
+        output_category="Feed",
+    )
+    exports.add(
+        obj=fs.feed.flow_mass_comp[0, "organic_solid"],
+        name="Organics(s) concentration",
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]:
```suggestion
        name="Organics(s) mass flow",
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+        ui_units=pyunits.metric_ton / pyunits.day,
+        display_units="ton/day",
+        rounding=1,
+        description="Inlet organics(solid) concentration",
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]:
```suggestion
        description="Inlet organics(solid) mass flow",
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+        is_input=True,
+        input_category="Feed",
+        is_output=True,
+        output_category="Feed",
+    )
+    exports.add(
+        obj=fs.feed.flow_mass_comp[0, "inorganic_solid"],
+        name="Inorganics(s) concentration",
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]: same suggestion as above: concentration --> mass flow

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py [hunk 7]
```diff
@@ -0,0 +1,641 @@
...
+        input_category="AT-HTL",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.ATHTL.energy_electric_flow_mass,
+        name="Electricity specific power",
+        ui_units=pyunits.kWh / pyunits.metric_ton,
+        display_units="kWh/ton of inlet flow rate",
+        rounding=2,
+        description="Specific power relating the power to inlet mass flow rate",
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]:
```suggestion
        description="Specific energy consumption with respect to influent mass",
```

**Review by adam-a-a — COMMENTED** [review]
Nice job with this! I just think there are a few typos in the naming/description that you should double-check before I re-review. I didn't get to scan the whole thing, so please double-check everything.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py (hunk 2 above)
Comment (luohezhiming) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/supercritical_sludge_to_gas_ui.py]: Yeah, it should be `organic_liquid`

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/tests/test_supercritical_sludge_to_gas.py [hunk 8]
```diff
@@ -101,10 +101,10 @@ def test_costing(self, system_frame):
...
 
         # check costing
-        assert value(m.fs.costing.LCOW) == pytest.approx(3537.28, rel=1e-3)  # in $/m**3
-        assert value(m.fs.costing.LCOG) == pytest.approx(23.7924, rel=1e-3)  # in $/kg
-        assert value(m.fs.costing.LCOC) == pytest.approx(407.846, rel=1e-3)  # in $/kg
-        assert value(m.fs.costing.LCOS) == pytest.approx(47.376, rel=1e-3)  # in $/kg
+        assert value(m.fs.costing.LCOW) == pytest.approx(466.187, rel=1e-3)  # in $/m**3
+        assert value(m.fs.costing.LCOG) == pytest.approx(3.1357, rel=1e-3)  # in $/kg
+        assert value(m.fs.costing.LCOC) == pytest.approx(53.751, rel=1e-3)  # in $/kg
+        assert value(m.fs.costing.LCOS) == pytest.approx(6.2438, rel=1e-3)  # in $/kg
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/supercritical_sludge_to_gas/tests/test_supercritical_sludge_to_gas.py:107]: I suppose there could be a check for LCOT too.

**Review by adam-a-a — APPROVED** [review]
LGTM - minor comment but I think we can merge and modify if needed in subsequent PRs. Great job!

**Pull request merged**